Google.Api.Gax.Grpc/Rest/RestChannel.cs

```diff
@@ -64,26 +64,17 @@ public RestChannel(RestServiceCollection serviceCollection, string endpoint, Cha
         {
             var restMethod = _serviceCollection.GetRestMethod(method);
 
-            var cancellationTokenSource = new CancellationTokenSource();
-            if (options.Deadline.HasValue)
-            {
-                // TODO: [virost, 2021-12] Use IClock.
-                var delayInterval = options.Deadline.Value - DateTime.UtcNow;
-                if (delayInterval.TotalMilliseconds <= 0)
-                {
-                    throw new RpcException(new Status(StatusCode.DeadlineExceeded, $"The timeout was reached when calling a method `{restMethod.FullName}`"));
-                }
-                cancellationTokenSource = new CancellationTokenSource(delayInterval);
-            }
+            var deadlineToken = CreateDeadlineCancellationToken(options, method.FullName);
+            var methodDisposalCts = new CancellationTokenSource();
 
-            var httpResponseTask = SendAsync(restMethod, host, options, request, cancellationTokenSource.Token, HttpCompletionOption.ResponseContentRead);
+            var httpResponseTask = SendAsync(restMethod, host, options, request, deadlineToken, HttpCompletionOption.ResponseContentRead);
             var readResponseTask = ReadResponseAsync(httpResponseTask);
```

Comment:
I think the linked token built by SendAsync should be passed to ReadResponseAsync as well and used to wrap the content reading call with something similar to AwaitWithCancellation.

Reply:
OK, but I still think that methodDisposalCts.Token should be passed to ReadResponseAsync (which currently expects no token), so the task that actually reads the content is cancallable as well.